HISTORY.md

```diff
@@ -1,6 +1,142 @@
 ---
 title: Release History
 ---
+# 3.0.0
+## Highlights
+Mesa 3.0 brings major improvements to agent-based modeling, making it more intuitive and powerful while reducing complexity. This release modernizes core functionalities and introduces new capabilities for both beginners and advanced users.
+
+### Streamlined agent management
+
+The centerpiece of Mesa 3.0 is its new agent management system. Agents are now automatically tracked and assigned unique IDs, eliminating common boilerplate code. The new AgentSet functionality provides an elegant and flexible way to work with agents:
+
+```python
+# Find agents meeting specific criteria
+wealthy_agents = model.agents.select(lambda a: a.wealth > 1000)
+
+# Group and analyze agents
+grouped = model.agents.groupby("species")
+species_stats = grouped.agg({
+    "count": len,
+    "avg_age": ("age", np.mean),
+    "total_wealth": ("wealth", sum)
+})
+
+# Activate agents in different patterns
+model.agents.shuffle_do("step")  # Random activation
+model.agents.select(lambda a: a.energy > 0).do("move")  # Conditional activation
+```
+
+The AgentSet provides powerful methods for filtering, grouping, and analyzing agents, making it easier to express complex model logic. Each model automatically maintains an AgentSet containing all agents (`model.agents`) and separate AgentSets for each agent type (`model.agents_by_type`).
+
+### Modern Visualization with SolaraViz
+
+Mesa 3.0's new visualization system, SolaraViz, provides a modern, interactive interface for model exploration:
```

Comment:
do we want to add here something about its work in progress status?

Reply:
Yes, fair.